File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/IsyGravingSourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class IsyGravingSourceApplication extends BaseSourceApplication {
    public IsyGravingSourceApplication(
            @Value("${fint.flyt.isygraving.sso.client-id:#{null}}") String clientId,
            @Value("${fint.flyt.isygraving.available:true}") boolean available
    ) {
        super(7L, "ISY Graving", clientId, available);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/IsyGravingSourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class IsyGravingSourceApplication(
    @Value("\${fint.flyt.isygraving.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.isygraving.available:false}")
    available: Boolean,
) : BaseSourceApplication(
        id = 7L,
        displayName = "ISY Graving",
        clientId = clientId,
        available = available,
    )
